vyper/builtin_functions/functions.py

```diff
@@ -1057,14 +1127,14 @@ def build_IR(self, expr, args, kwargs, context):
 class RawCall(_SimpleBuiltinFunction):
 
     _id = "raw_call"
-    _inputs = [("to", AddressDefinition()), ("data", BytesArrayPrimitive())]
+    _inputs = [("to", AddressDefinition()), ("data", BytesAbstractType())]
     _kwargs = {
-        "max_outsize": Optional("num_literal", 0),
-        "gas": Optional("uint256", "gas"),
-        "value": Optional("uint256", zero_value),
-        "is_delegate_call": Optional("bool", false_value),
-        "is_static_call": Optional("bool", false_value),
-        "revert_on_failure": Optional("bool", true_value),
+        "max_outsize": Optional(Uint256Definition(), 0, require_literal=True),
+        "gas": Optional(Uint256Definition(), "gas"),
+        "value": Optional(Uint256Definition(), zero_value),
+        "is_delegate_call": Optional(BoolDefinition(), false_value),
```

Comment:
not sure if this is out of scope, but these three should all probably have `require_literal=True`.

Reply:
i checked, and using `require_literal=True` will simplify the rest of the implementation of RawCall, so i think it's worth doing here.

Reply:
I noticed that `examples/wallet/wallet.vy` takes in a non-literal. Are you referring to `is_delegate_call`, `is_static_call` and `revert_on_failure` instead? There is a check in `build_IR` that these are literals.